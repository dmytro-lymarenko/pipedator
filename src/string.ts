import { createValidator } from './core';

export function string() {
	return createValidator({
		message: 'Value should be a string',
		validate: (value, ctx) => (typeof value === 'string' ? null : { value, message: ctx.message, path: ctx.path }),
	});
}

export function empty() {
	return createValidator({
		message: 'Value should be empty',
		validate: (value, ctx) => (value.length === 0 ? null : { value, message: ctx.message, path: ctx.path }),
	});
}
