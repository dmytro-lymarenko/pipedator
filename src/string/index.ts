import { Validator } from '../core';
import { test } from '../test';

function isString(value: any): value is string {
	return typeof value === 'string';
}

export function string(/* ...validators: Validator<string>[] */): Validator<string> {
	return test<string>(isString, 'Value should be a string');
}
