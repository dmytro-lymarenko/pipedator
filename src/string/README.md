- **Description**

Creates a validator to check if a value is a string or not.

- **How to import**

```typescript
import { string } from 'pipedator';
// or
import { string } from 'pipedator/lib/string';

```
- **Signature**

```typescript
function string<ValidValue = any>(message?: string): Validator<ValidValue>;
```
- **Parameters**

  - `message` - (optional) custom message.


- **Usage**

```typescript
string().validate('Some text'); // valid
string().validate(5); // invalid
string('Custom message').validate(null); // invalid with 'Custom message'
```