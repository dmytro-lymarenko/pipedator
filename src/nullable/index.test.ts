import { nullable } from './index';
import { string } from '../string';

describe('nullable()', () => {
	it('should validate a string', () => {
		expect(nullable(string()).validate('Some test have passed')).toBeNull();
	});

	it('should validate a string when it is actually null', () => {
		expect(nullable(string()).validate(null)).toBeNull();
	});

	it('should fail', () => {
		expect(nullable(string()).validate(1)).toMatchSnapshot();
	});
});
