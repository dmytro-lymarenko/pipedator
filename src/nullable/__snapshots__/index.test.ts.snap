// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nullable() should fail 1`] = `
Object {
  "children": null,
  "message": "Either Value should equal to null OR Value should be a string",
  "path": "",
}
`;
